File: my-folder/0011-container-with-most-water/solution.java
class Solution {
    public int maxArea(int[] height) {
        int res = 0;
        int left = 0;
        int right = height.length - 1;
        while (left < right) {
            int w = right - left;
            int h = Math.min(height[right], height[left]);
            int water = w*h;
            res = Math.max(res, water);
            if (height[left] >= height[right]) {
                right--;
            }
            else {
                left++;
            }
        } 
        return res;
    }
}

File: my-folder/0011-container-with-most-water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        max_water = 0
        left = 0
        right = len(height) - 1

        while left < right:
            water = (right - left) * min(height[left], height[right])

            max_water = max(max_water, water)

            if (height[left] >= height[right]):
                right -= 1
            
            else:
                left += 1
            
        return max_water
